src/org/lwjgl/demo/opengl/SimpleDrawElements.java

```diff
@@ -117,15 +119,28 @@ private void loop() {
         // Set the clear color
         glClearColor(1.0f, 0.0f, 0.0f, 0.0f);
 
+        // Generate vertex and index buffers
         int vbo = glGenBuffers();
         int ibo = glGenBuffers();
+
+        // Vertex and index data
         float[] vertices = {-0.5f, -0.5f, 0.5f, -0.5f, 0.5f, 0.5f};
         int[] indices = {0, 1, 2};
-        glBindBuffer(GL_ARRAY_BUFFER, vbo);
-        glBufferData(GL_ARRAY_BUFFER, (FloatBuffer) BufferUtils.createFloatBuffer(vertices.length).put(vertices).flip(), GL_STATIC_DRAW);
-        glEnableClientState(GL_VERTEX_ARRAY);
-        glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ibo);
-        glBufferData(GL_ELEMENT_ARRAY_BUFFER, (IntBuffer) BufferUtils.createIntBuffer(indices.length).put(indices).flip(), GL_STATIC_DRAW);
+
+        // Load the data into buffers and send it off to OpenGL using the MemoryStack api
+        // instead of BufferUtils
+        // Note: Don't forget to flip any buffers before passing them into gl functions that need them
+        try ( MemoryStack stack = MemoryStack.stackPush() ) {
```

Comment:
I think using the MemoryStack for buffer data upload is stretching it a bit. In this example, since it is only 3 vertices it's okay, but it might give the wrong idea to people. The problem is that MemoryStack is not meant for potentially large buffer data, which one usually has when uploading models/vertices. A better alternative for this is manual Buffer memory management with memAlloc/memFree.

Reply:
I had forgotten to state why I used the MemoryStack vs BufferUtils. From what I know the use of the BufferUtils class is discouraged. I'll add a note about using different methods for allocating buffers.